packages/core/src/processors/tool-call-filter-utils.ts

```
@@ -0,0 +1,735 @@
+          const remaining = maxTextCodeUnits === undefined ? undefined : Math.max(0, maxTextCodeUnits - textLength);
+          const bounded = remaining === undefined ? value : value.slice(0, remaining);
+          text.push(bounded);
+          textLength += bounded.length;
+          if (maxTextCodeUnits !== undefined && textLength >= maxTextCodeUnits) break;
+        }
+        return text.length > 0 ? text.join('') : null;
+      }
+      default:
+        // Preserve the two legacy wrapper shapes ToolCallFilter already accepted, but never stringify
+        // arbitrary objects. This fails closed for media and unknown provider-specific output.
+        const textProperty = getOwnDataProperty(output, 'text');
```

Biome (2.5.1)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)